### PR TITLE
FSET-706 Groundwork: Add the ability to hard code firstnames, lastnames and preferred names in the test data generator

### PR DESCRIPTION
**Pull request by henricook**

### CODE REVIEW

File: app/services/testdata/RegisteredStatusGenerator.scala [hunk 1]
```diff
@@ -24,41 +24,38 @@ import services.testdata.faker.DataFaker._
...
+  override val authProviderClient = AuthProviderClient
 }
 
 trait RegisteredStatusGenerator extends BaseGenerator {
   import scala.concurrent.ExecutionContext.Implicits.global
 
-  val appRepository: GeneralApplicationRepository
-  val pdRepository: PersonalDetailsRepository
-  val qRepository: QuestionnaireRepository
+  val authProviderClient: AuthProviderClient.type
```
Comment (kjharkin) [inline app/services/testdata/RegisteredStatusGenerator.scala]: Just a question: interesting use of Singleton.type here. What was the intention?

Comment (henricook) [inline app/services/testdata/RegisteredStatusGenerator.scala]: I think i've used it previously when injecting things that exist as singletons within the app. If I take away the .type things don't work as expected, not an overly scientific explanation! - how would you do this?

**Comment by kjharkin** [comment]
Looks good.

File: app/services/testdata/RegisteredStatusGenerator.scala (hunk 1 above)
Comment (kjharkin) [inline app/services/testdata/RegisteredStatusGenerator.scala]: I'm not sure what the issue is if I'm being honest. Does one have to specify Singleton.type when injecting any singleton or only when one wants to limit an interface (trait) to a specific Singleton? Someone I'll need to try.

Comment (henricook) [inline app/services/testdata/RegisteredStatusGenerator.scala]: I think it's whenever you inject a Singleton, which we don't really do much. I can't find many links to support it though from a cursory look, maybe i'll have a proper google later

**Pull request merged**